src/chart/generator/marker.cpp

```diff
@@ -75,7 +76,12 @@ Marker::Marker(const Options &options,
 	    index,
 	    horizontal ? &mainId->value : subAxisId);
 
-	spacing.x = (horizontal || (lineOrCircle && !polar))
+	spacing.x = (horizontal || (lineOrCircle && !polar)
+	                || (channels.at(ChannelId::y).isDimension()
+	                    && channels.at(ChannelId::y).hasDimension()
```

Comment:
Why do we need hasDimension(), shouldnt isDimension be enought?

Reply:
isDimension() && !hasDimension means == empty(). When it is an empty channel, there should be no spacing